### ISSUE TITLE
Invalid Hook Call

### ISSUE THREAD

**Issue by rehanumar** [rcdexta/react-trello#453]
**Describe the bug**
`react-trello` version `2.2.*` is breaking the rules of hook with `react` and `react-dom` version 16.13.1`. App is crashing in this case.

**To Reproduce**
Create a new project with `react` and `react-dom` version `16.13.1` and install `react-trello`.
(https://github.com/rehanumar/react-trello-bug-reproduce)[https://github.com/rehanumar/react-trello-bug-reproduce]

**Expected behavior**
React-trello should not throw error and should render page properly.

**Screenshots**
![https://drive.google.com/file/d/1ZJEH7BqH5N7WlO6ObszUMwlYtkM_rqcA/view?usp=sharing](screenshot)

PS: I am not sure whether this is the right repo to discus this bug or I might need to post that with some upstream library of react-trello. I have search react-trello codebase from version `2.2.4` to `2.2.10`. None of them were using hooks.

Context:
Basically, everything was working correctly on my local computer. Until I deleted the `node_modules` and re installed the version `2.2.10` of `react-trello` to replicate this issue happening on staging.

**Comment by urosran** [comment]
@dapi @rcdexta this issue still persists - any updates as to when can it be fixed?

**Comment by Mohamed-Ayaan358** [comment]
Issue still persists at 17.0.2 unaware of how to solve it

**Comment by marciok** [comment]
Make sure your `react` dependency is the same as `react-dom`, ex:

package.json:
```
    "react": "17.0.2",
    "react-dom": "17.0.2",
```

**Comment by kocka996** [comment]
I get this error. Is there any way to resolve this?
 "react": "^17.0.2",
 "react-dom": "^17.0.2"

**Comment by lucioperca** [comment]
I was able to resolve a similar error by forcing react-trello to use a pre-v6 version of styled-components. Basically, throwing this into package.json:

```
  "resolutions": {
    "react-trello/**/styled-components": "5.3.11"
  }
```

I'm not 100% convinced this is the same problem as encountered above, but maybe it will help someone else who runs across this issue.
